src/dmd/lexer.d

```diff
@@ -202,6 +202,129 @@ unittest
     }
 }
 
+
+/**************
+ * Pool allocator for Token objects, used by the Lexer.
+ * Tokens are never freed so Token pointers will never change
+ */
+private struct TokenPool{
+private:
+    enum PoolSize = 1024; //number of tokens per pool
+    enum DefaultPoolListSize = 64; //number of pools to allocate by default
+    Token[][] pools;
```

Comment:
Just use an `Array` - the code here is re-inventing what `Array` does.

Reply:
I chose this representation to avoid copying tokens, but I'm going to experiment to see if a 1D array is actually better.  For now, I updated this to use Array!(Token[])